mediapipe/tasks/python/vision/core/base_vision_task_api.py

```diff
@@ -122,6 +127,50 @@ def _send_live_stream_data(self, inputs: Mapping[str, _Packet]) -> None:
           + self._running_mode.name)
     self._runner.send(inputs)
 
+  @staticmethod
+  def convert_to_normalized_rect(
+      options: _ImageProcessingOptions,
+      roi_allowed: bool = True
+  ) -> _NormalizedRect:
+    """
+    Convert from ImageProcessingOptions to NormalizedRect, performing sanity
+    checks on-the-fly. If the input ImageProcessingOptions is not present,
+    returns a default NormalizedRect covering the whole image with rotation set
+    to 0. If 'roi_allowed' is false, an error will be returned if the input
+    ImageProcessingOptions has its 'region_of_interest' field set.
+
+    Args:
+      options: Options for image processing.
+      roi_allowed: Indicates if the `region_of_interest` field is allowed to be
+        set. By default, it's set to True.
+
+    """
+    normalized_rect = _NormalizedRect(rotation=0, x_center=0.5, y_center=0.5,
+                                      width=1, height=1)
+    if options is None:
+      return normalized_rect
+
+    if options.rotation_degrees % 90 != 0:
+      raise ValueError("Expected rotation to be a multiple of 90°.")
+
+    # Convert to radians counter-clockwise.
+    normalized_rect.rotation = -options.rotation_degrees * math.pi / 180.0
+
+    if options.region_of_interest:
+      if not roi_allowed:
+        raise ValueError("This task doesn't support region-of-interest.")
+      roi = options.region_of_interest
+      if roi.x_center >= roi.width or roi.y_center >= roi.height:
+        raise ValueError(
+          "Expected Rect with x_center < width and y_center < height.")
+      if roi.x_center < 0 or roi.y_center < 0 or roi.width > 1 or roi.height > 1:
+        raise ValueError("Expected Rect values to be in [0,1].")
+      normalized_rect.x_center = roi.x_center + roi.width / 2.0
+      normalized_rect.y_center = roi.y_center + roi.height / 2.0
+      normalized_rect.width = roi.width - roi.x_center
+      normalized_rect.height = roi.height - roi.y_center
```

Comment:
Feel like there are some mis-naming here.
https://github.com/kinaryml/mediapipe/blame/a913255080b10692808a9a66edd10f1e490758af/mediapipe/tasks/python/components/containers/rect.py#L30-L36
The definition of rect.x_center and rect.y_center is misleadning.
It should be something like this if rect is using center coordinates:
```suggestion
      normalized_rect.x_center = roi.x_center
      normalized_rect.y_center = roi.y_center
      normalized_rect.width = roi.width
      normalized_rect.height = roi.height
```

If using top-left coordinates for rect, should be
```suggestion
      normalized_rect.x_center = roi.x_center + roi.width / 2.0
      normalized_rect.y_center = roi.y_center + roi.height / 2.0
      normalized_rect.width = roi.width
      normalized_rect.height = roi.height
```

Reply:
Uses the top-left coordinates for Rect now (same as Java/C++)